Sources/SwiftGraphQLClient/Extensions/Publishers+Extensions.swift

```diff
@@ -8,6 +8,42 @@ extension Publisher {
     func takeUntil<Terminator: Publisher>(_ terminator: Terminator) -> Publishers.TakenUntilPublisher<Self, Terminator> {
         Publishers.TakenUntilPublisher<Self, Terminator>(upstream: self, terminator: terminator)
     }
+
+    /// Takes the first emitted value and completes or throws an error
+    func first() async throws -> Output {
+        try await withCheckedThrowingContinuation { continuation in
+            var cancellable: AnyCancellable?
+
+            cancellable = first()
+                .sink { result in
+                    switch result {
+                    case .finished:
+                        break
```

Comment:
I'm not sure but seems like the publisher could finish without value. Would the continuation live forever in that case?

Reply:
> I'm not sure but seems like the publisher could finish without value. Would the continuation live forever in that case?

Yeap, I also not sure about this implementation. But according to the documents:

> CheckedContinuation performs runtime checks for missing or multiple resume operations

So, we at least should get a runtime warning if it completes without values